alica_common_config/include/alica_common_config/debug_output.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+namespace alica
+{
+
+enum AlicaDebugLevel
+{
+    None = 0,
+    Error,
+    Warning,
+    Info,
+    Debug,
+    All
+};
+
+#ifdef ALICA_DEBUG_LEVEL_ALL
+#define ALICA_DEBUG_LEVEL AlicaDebugLevel::All
+#elif defined ALICA_DEBUG_LEVEL_DEBUG
+#define ALICA_DEBUG_LEVEL AlicaDebugLevel::Debug
+#elif defined ALICA_DEBUG_LEVEL_INFO
+#define ALICA_DEBUG_LEVEL AlicaDebugLevel::Info
+#elif defined ALICA_DEBUG_LEVEL_WARNING
+#define ALICA_DEBUG_LEVEL AlicaDebugLevel::Warning
+#elif defined ALICA_DEBUG_LEVEL_ERROR
+#define ALICA_DEBUG_LEVEL AlicaDebugLevel::Error
+#endif
+
+#ifndef ALICA_DEBUG_LEVEL
+#define ALICA_DEBUG_LEVEL AlicaDebugLevel::None
+#else
+#include <iostream>
+#endif
+
+#define ALICA_ERROR_MSG(message)                                                                                                                               \
+    do {                                                                                                                                                       \
```

Comment:
why do ... while(0), why not just enclose in brackets if its for scope block

Reply:
so that the meaning of the semicolon behind the statement is clear.

https://www.quora.com/What-is-the-purpose-of-using-do-while-0-in-macros